File: q1004/q1004.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int res = 0;
        int start = 0;
        int flipped = 0;
        for (int end = 0; end < nums.length; end++) {
            if (nums[end] == 0)
                flipped++;

            while (flipped > k) {
                if (nums[start] == 0)
                    flipped--;
                start++;
            }

            res = Math.max(res, end - start + 1);
        }
        return res;
    }
}

File: q1004/q1004.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int zero = 0; // to keep track of the number of zeroes in our current window
        int max_len = INT_MIN, l = 0;
        for(int r = 0; r < nums.size(); r ++){
            if(nums[r] == 0)zero ++;
            // if the number of zeroes exceeds k, then shrink the window
            while(zero > k){
                if(nums[l] == 0){
                    zero --;
                }
                l ++;
            }
            max_len = max(max_len, r - l + 1);
        }
        return max_len;
    }
};